mapper/static/js/mapper_generate.js

```diff
@@ -9,16 +9,34 @@ Mapper.generate = new function() {
 			.setView([38.1089, 13.3545], 8)
 			.addControl(L.mapbox.geocoderControl('mapbox.places'));
 
-		Mapper.map_components.activeMarkerLayer = new L.mapbox.featureLayer().addTo(Mapper.map_components.map);
-
-
+		if (Mapper.map_id) {
+			Mapper.generate.loadMap()
+		}
+		else {
+			Mapper.map_components.activeMarkerLayer = new L.mapbox.featureLayer()
+				.addTo(Mapper.map_components.map);
+		}
 		Mapper.map_components.storageMarkerLayer = new L.geoJson().addTo(Mapper.map_components.map);
+
 		Mapper.map_components.storageMarkerLayer.options.pointToLayer = L.mapbox.marker.style;
 
 		Mapper.generate.addListeners();
 
 	}
 
+	this.loadMap = function() {
+			Mapper.map_components.activeMarkerLayer = new L.mapbox.featureLayer()
```

Comment:
You actually want to add all these points to the `Mapper.map_components.storageMarkerLayer`. The active layer is reserved for a single marker that's being created/edited.